File: python_leaoai/models/Factory/ProjectItem.py
# models/project_item.py

from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship
from datetime import datetime
from typing import List
from database import Base
from .project import Project
from .project_phase import ProjectPhase

class ProjectItem(Base):
    __tablename__ = "project_item"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)

    project_id: Mapped[int] = mapped_column(ForeignKey("project.id"))
    project: Mapped["Project"] = relationship("Project")

    name: Mapped[str] = mapped_column(String, nullable=False)
    pdf_report: Mapped[str] = mapped_column(String, nullable=True)
    product_name: Mapped[str] = mapped_column(String, nullable=True)
    acabamento: Mapped[str] = mapped_column(String, nullable=True)
    codigo: Mapped[str] = mapped_column(String, nullable=True)
    amount: Mapped[int] = mapped_column(Integer, nullable=True)
    
    is_approved: Mapped[bool] = mapped_column(Boolean, default=False)
    is_reproved: Mapped[bool] = mapped_column(Boolean, default=False)
    concluded_at: Mapped[DateTime] = mapped_column(DateTime, nullable=True)
    started_at: Mapped[DateTime] = mapped_column(DateTime, nullable=True)

    ativo: Mapped[bool] = mapped_column(Boolean, default=True)
    excluido: Mapped[bool] = mapped_column(Boolean, default=False)
    is_template: Mapped[bool] = mapped_column(Boolean, default=False)

    acabamento_id: Mapped[int] = mapped_column(Integer, nullable=True)
    embalagem_caixa_id: Mapped[int] = mapped_column(Integer, nullable=True)
    embalagem_papelao_id: Mapped[int] = mapped_column(Integer, nullable=True)
    embalagem_fechamento_id: Mapped[int] = mapped_column(Integer, nullable=True)

    project_phases: Mapped[List["ProjectPhase"]] = relationship(
        "ProjectPhase", back_populates="project_item"
    )

File: Framework.Data/Models/Factory/ProjectItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Framework.Data.Models.Factory
{
	// PROJECT ITEM
	public class ProjectItem
	{
		[Key]
		public int Id { get; set; }

		[ForeignKey("Project")]
		public int ProjectId { get; set; }
		public virtual Project Project { get; set; }

		[Required]
		public string Name { get; set; }

		public string PdfReport { get; set; }
        public string ProductName { get; set; }
        public string Acabamento { get; set; }
        public string Codigo { get; set; }
        public int? Amount { get; set; }
		public bool IsApproved { get; set; } = false;
		public bool IsReproved { get; set; } = false;
		public DateTime? ConcludedAt { get; set; }
		public DateTime? StartedAt { get; set; }

		public bool Ativo { get; set; } = true;
		public bool Excluido { get; set; } = false;
		public bool IsTemplate { get; set; } = false;
        public int? AcabamentoId { get; set; }
        // FK para Embalagem (tipo Caixa)
        public int? EmbalagemCaixaId { get; set; }

        // FK para Embalagem (tipo Papelão)
        public int? EmbalagemPapelaoId { get; set; }

        // FK para Embalagem (tipo Fechamento)
        public int? EmbalagemFechamentoId { get; set; }

        // Navegação
        public virtual ICollection<ProjectPhase> ProjectPhases { get; set; } = new List<ProjectPhase>();
	}
}
